src/main/java/task/TodoWithinPeriod.java

```diff
@@ -0,0 +1,33 @@
+package task;
+
+import parser.DateTimeExtractor;
+
+import java.io.Serializable;
+import java.time.LocalDateTime;
+
+public class TodoWithinPeriod extends Todo implements Serializable {
+
+    /**
+     * Creates a ToDo task with a specific duration and timing.
+     *
+     * @param description description of task
+     * @param startDate   start time of the task
+     * @param endDate     end time of the task
+     */
+    public TodoWithinPeriod(String description, LocalDateTime startDate, LocalDateTime endDate) {
+        super(description);
+        this.startDate = startDate;
+        this.endDate = endDate;
```

Comment:
What is the purpose of having so many new classes with one additional attribute rather than having it as an attribute in Todo with default Null value?